src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -86,6 +88,198 @@ keeper_pg_init_fsm(Keeper *keeper, KeeperConfig *config)
+		join_path_components(defaultIdentConfPath, defaultConfigurationPath, "pg_ident.conf");
+
+		move_file(defaultIdentConfPath, identConfPath);
+	}
+
+	/* disable auto start in default configuration */
+	disableAutoStart(defaultConfigurationPath);
+
+	return true;
+}
+
+
+/*
+ * disable_configuration_parameters read postgresql.conf file and
+ * disables predefined configuration parameters inside.
+ */
+static bool
+disable_configuration_parameters(char *postgresqlConfPath)
+{
+	FILE *fileStream = NULL;
+	PQExpBuffer newConfContents = NULL;
+	char lineBuffer[BUFSIZE];
+	/*
+	 * configuration parameters can appear in any order, and we
+	 * need to check for patterns for NAME = VALUE and NAME=VALUE
+	 */
+	char *targetVariables[] = {
+			"data_directory ", "data_directory =",
+			"hba_file ", "hba_file=",
+			"ident_file ", "ident_file=",
+			"include_dir ", "include_dir=",
+			NULL};
+
+	/* open a file */
+	fileStream = fopen(postgresqlConfPath, "rb");
+	if (fileStream == NULL)
+	{
+		log_error("Failed to open file \"%s\": %s", postgresqlConfPath, strerror(errno));
+		return false;
+	}
+
+	newConfContents = createPQExpBuffer();
+	if (newConfContents == NULL)
+	{
+		log_error("Failed to allocate memory");
+		return false;
+	}
+
+	/* read each line including terminating new line and process it*/
+	while (fgets(lineBuffer, BUFSIZE, fileStream) != NULL)
+	{
+		bool variableFound = false;
+		int targetIndex = 0;
+		char *targetVariable = targetVariables[0];
+
+		/* check if the line contains any of target variables */
+		while (targetVariable != NULL)
+		{
+			char *previousOffset = NULL;
+			char *candidateOffset = strstr(lineBuffer, targetVariable);
+			if (candidateOffset != NULL)
+			{
+				variableFound = true;
+				break;
+			}
+			targetVariable = targetVariables[++targetIndex];
+		}
+
+		/*
+		 * disable the line if any of target variables is found
+		 * and if it was not already disabled
+		 */
+		if (variableFound && lineBuffer[0] != '#')
+		{
+			appendPQExpBufferChar(newConfContents, '#');
+		}
+
+		/* copy rest of the line */
+		appendPQExpBufferStr(newConfContents, lineBuffer);
+	}
+
+	fclose(fileStream);
+
+	write_file(newConfContents->data, newConfContents->len, postgresqlConfPath);
+
+	destroyPQExpBuffer(newConfContents);
+
+	return true;
+}
+
+
+/* disableAutoStart disables auto start in default configuration */
+static void
+disableAutoStart(char *defaultConfigurationPath)
+{
+	char defaultStartConfPath[MAXPGPATH];
+	char newStartConfPath[MAXPGPATH];
+	char *newStartConfData = "disabled";
+	char *fileContents = NULL;
+	long fileSize = 0;
+
+	join_path_components(defaultStartConfPath, defaultConfigurationPath, "start.conf");
+	join_path_components(newStartConfPath, defaultConfigurationPath, "start.conf.orig");
+
+	if (read_file(defaultStartConfPath, &fileContents, &fileSize))
+	{
+		write_file(fileContents, fileSize, newStartConfPath);
+	}
+
+	write_file(newStartConfData, strlen(newStartConfData), defaultStartConfPath);
```

Comment:
It would be nice to add a comment that the file has been edited by `pg_auto_failover`, as we do in all the configuration files we edit.